File: src/lecture1/lecture1_my_topic_py/scripts/subscriber.py
#!/usr/bin/env python3
import rospy
from lecture1_msgs.msg import TimeUInt32

def callback(msg):
    rospy.loginfo("%i.%i : %d", msg.stamp.secs, msg.stamp.nsecs, msg.data)

def subscriber():
    rospy.init_node("subscriber", anonymous=True)
    rospy.Subscriber("topic", TimeUInt32, callback)
    rospy.spin()

if __name__ == '__main__':
    subscriber()

File: src/lecture1/lecture1_my_topic_cpp/src/subscriber.cpp
#include <ros/ros.h>
#include <lecture1_msgs/TimeUInt32.h>

void callback(const lecture1_msgs::TimeUInt32::ConstPtr& msg)
{
    ROS_INFO_STREAM(msg->stamp << " : " << msg->data);
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "subscriber", ros::init_options::AnonymousName);
    ros::NodeHandle n;
    ros::Subscriber s = n.subscribe("topic", 5, callback);
    ros::spin();
    return 0;
}

